pkg/provisioningv2/rke2/planner/s3args.go

```diff
@@ -88,11 +88,11 @@ func getS3Credential(secretCache corecontrollers.SecretCache, namespace, name, r
 	}
 
 	secret, err := machineprovision.GetCloudCredentialSecret(secretCache, namespace, name)
-	if err != nil {
-		return result, fmt.Errorf("failed to lookup etcdSnapshotCloudCredentialName: %w", err)
-	}
-	if secret.Type != "provisioning.cattle.io/cloud-credential" && secret.Namespace != namespaces.GlobalNamespace {
-		return result, fmt.Errorf("expected secret of type provisioning.cattle.io/cloud-credential, got [%s]", secret.Type)
+	if err != nil || (secret.Type != "provisioning.cattle.io/cloud-credential" && secret.Namespace != namespaces.GlobalNamespace) {
```

Comment:
Per the discussion with @vincent99 can we remove the check for Type "provisioning.cattle.io/cloud-credential."  Basically as long as GetCloudCredentialSecret doesn't return an error use it.